src/Digdir.Domain.Dialogporten.Janitor/CostManagementAggregation/COST_METRICS_SETUP.md

```diff
@@ -0,0 +1,161 @@
+# Cost Metrics Aggregation - Setup Guide
+
+This document provides step-by-step instructions for setting up the cost metrics aggregation job in production.
+
+## Overview
+
+The cost metrics aggregation job runs in the production environment and:
+- Queries Application Insights from all environments (prod, staging, test, yt01)
+- Aggregates cost metrics data
+- Uploads results to a storage account in the prod resource group
+- Runs daily at 2:00 AM
+
+## Prerequisites
+
+- Access to Azure Portal
+- Access to the production Key Vault
+- Permissions to grant RBAC roles
+- The job must be deployed first (via GitHub Actions)
+
+## Step 1: Deploy the Job
+
+The job is automatically deployed via GitHub Actions when you deploy to the production environment. The deployment will:
+
+1. Create a user-assigned Managed Identity
+2. Create a storage account in the prod resource group
+3. Create a blob container named "costmetrics"
+4. Deploy the Container App Job
+
+## Step 2: Add Secrets to Key Vault
+
+Add the following secrets to the **production Key Vault**:
+
+| Secret Name | Description | Example Value |
+|-------------|-------------|---------------|
+| `aggregateCostMetricsStagingSubscriptionId` | Staging subscription ID | `12345678-1234-1234-1234-123456789abc` |
+| `aggregateCostMetricsProdSubscriptionId` | Production subscription ID | `87654321-4321-4321-4321-cba987654321` |
+| `aggregateCostMetricsTestSubscriptionId` | Test subscription ID | `11111111-2222-3333-4444-555555555555` |
+| `aggregateCostMetricsYt01SubscriptionId` | YT01 subscription ID | `99999999-8888-7777-6666-555555555555` |
+
+**Note:** These are the subscription IDs where the Application Insights resources are located, not the subscription where the job runs.
+
+## Step 3: Grant RBAC Permissions
+
+After deployment, you need to grant the Managed Identity read access to Application Insights in all environments.
+
+### Find the Managed Identity
+
+1. Go to Azure Portal
+2. Navigate to the production resource group
+3. Find the user-assigned managed identity: `dp-be-prod-aggregate-cost-metrics-identity`
+4. Copy the **Object (principal) ID**
+
+### Grant Application Insights Access
```

Comment:
this will be solved automatically now right? If not, should we add the `Monitoring Reader` role as well here? 

But that will probably be a bit hard because we need to have access to the staging/test subscription as well to add the correct permissions? 🤔 

If we redo it to run the job in the environment we don't have to add it manually.